### PR TITLE
Tweak Language Reference documentation

### PR DESCRIPTION
**Pull request by pikinier20**
- For sections that had `overview` page, changed the index of section to `overview` and created redirects so that link to old `overview.html` leads to `index.html`
- Added dummy index pages to sections that didn't have any index. The default content of index page is a table of child pages.
- Turned off snippet run feature for snippets that are not checked with snipper compiler or failed.
- Deleted version picker for reference documentation

### CODE REVIEW

File: docs/_layouts/index.html [hunk 1]
```diff
@@ -1,4 +1,15 @@
...
+{{ content }}
+
+<h2>Table of Contents</h2>
+<ul class="table-of-contents">
+  {% for child in site.posts %}
+      <li>
+          <a href="{{ child.url }}">{{ child.title }}</a>
+      </li>
+  {% endfor %}
+</ul>
```
Comment (julienrf) [inline docs/_layouts/index.html:15]: Great idea!

File: docs/_layouts/index.html [hunk 2]
```diff
@@ -1,4 +1,15 @@
...
+layout: static-site-main
 ---
-<h1>{{ content }}</h1>
+<h1>{{ page.title }}</h1>
+
+{{ content }}
+
+<h2>Table of Contents</h2>
+<ul class="table-of-contents">
+  {% for child in site.posts %}
```
Comment (julienrf) [inline docs/_layouts/index.html:10]: What is `site.posts`? It looks like it’s going to be the list of blog posts, but I guess it should be the list of subpages?

**Review by julienrf — COMMENTED** [review]
The diff looks good to me, but please share the PR online URL when it is ready so that we can do more checks.

File: docs/_layouts/index.html (hunk 2 above)
Comment (pikinier20) [reply docs/_layouts/index.html:10]: I'll rename this field to `subpages`. Until now, we used this field to generate blogs index and that's why it's called `posts`

**Comment by bishabosha** [comment]
> https://scala3doc.virtuslab.com/pr-tweak-reference-documentation/reference/index.html

I looked at this, shouldn't the more-details pages be hidden from contents/index?

**Review by julienrf — APPROVED** [review]
Apart from the point raised by Jamie, this looks great!

**Comment by pikinier20** [comment]
The `More details` pages were hidden but I revealed them at some point. Isn't it better to let user know that such page exists instead to have only one place on site where he can navigate to it?

**Comment by bishabosha** [comment]
> The `More details` pages were hidden but I revealed them at some point. Isn't it better to let user know that such page exists instead to have only one place on site where he can navigate to it?

I considered that the reason they did not appear in the nav-bar was because those pages are meant to be "part of" the page they are linked from, and its just incidental that they had a different url. So I advocate in this case for modal content, or "reveal" boxes or whatever - or if none of that then we should retitle the pages or just inline their content

**Comment by pikinier20** [comment]
> 

We can make another section on these pages called: More details and then put there the "more advanced" stuff.

**Comment by pikinier20** [comment]
@julienrf @bishabosha If the "More details" pages are the only concern, could we merge this PR and make another PR with the documentation changes? Perhaps you wanna take it over?

**Pull request merged**